core/src/main/java/com/linecorp/armeria/server/AbstractHttpResponseSubscriber.java

```diff
@@ -384,6 +384,9 @@ private void addCallbackAndFlush(Throwable cause, State oldState, ChannelFuture
                 try (SafeCloseable ignored = RequestContextUtil.pop()) {
                     if (f.isSuccess() && !isReset) {
                         maybeLogFirstResponseBytesTransferred();
+                        if (req.shouldResetIfRemoteIsOpen()) {
+                            responseEncoder.writeReset(req.id(), req.streamId(), Http2Error.CANCEL, false);
```

Comment:
Should be? Because `shouldResetIfRemoteIsOpen` is true.
```suggestion
                            responseEncoder.writeReset(req.id(), req.streamId(), Http2Error.CANCEL, true);
```